### ISSUE TITLE
Feature Request - Notification Sound While in Conversation

### ISSUE THREAD

**Issue by Endda** [DrKLO/Telegram#734]
I am trying to switch from WhatsApp to Telegram and one feature that I am missing is the ability to hear/set a notification sound while I am in the conversation. Sometimes I just like to put my phone down and when I get a message from the person I am actively talking to(and have the conversation up), I would like to hear a notification sound, or even a vibrate, when they message me.

Thank you for all the hard work you guys(and gals?) have put into making Telegram. I am trying to convince my girlfriend to switch from WhatsApp and this is one feature that we seem to be missing. I could probably live without it but it would definitely be nice to see implemented.

Thanks again! :)

**Comment by Darkwood77** [comment]
I would very much like to second this. I found another thread where DrKlo said that notification sound when in chat that is already open is disabled, as it is *probably* not needed, but both me and 3 of my friends that migrated from Whatsapp to Telegram would dearly like to have this.

Additionaly, we have checked this functionality on iPhone - there it is working exactly like on Whatsapp, i.e. there is a small, quieter chime when a message is received in an open chat, rather than a loud standard beep when a different chat message is received or when Telegram is closed.

But so many times it has happenned that I haven't at all noticed a new message in an open chat due to the lack of sound. At least having an option to switch it on/off would be grand.

Cheers

**Comment by Alirezaaa** [comment]
notifications are just for the time that You're not in that window, but while You are in a chat, what is the necessary to have notifications from the chat window?

**Comment by Endda** [comment]
Because not everyone stares at their phone while they're talking to someone. We all use our smartphones and tablets differently than everyone else. Don't try to say that because you don't think it's useful then no one else in the world would like the feature.

As you can see, people are trying to move away from WhatsApp but when integral features like this are missing, it makes it very hard to convince our friends to switch.

With Telegram fighting an uphill battle of trying to get as many people to switch from WhatsApp(and really any other messaging app), these missing features are important.

**Comment by Darkwood77** [comment]
@Alirezaaa basically what Endda said. And I can illustrate with a few examples:

-Quite a few times already I have noticed that while I was typing a message in an open chat and had received another one in the meantime, I didn't even notice it as my visual concentration was mostly focused on what I was typing, not the chat above. It was only after I sent my own message that I had noticed a couple of new ones popped up there. If there was a small quite chime alerting me that these messages have appeared, I would have noticed them.

-My friend has a very long screen timeout, he tells me that in a fairly active chat on whatsapp he will leave the conversation screen open as he doesn't want the default loud notification beep over and over again, so he just picks his phone up when he hears the quiet chime of a new message arriving. He cannot do this in Telegram as new messages to the open chat do not have sound notification. 

-I myself have often noticed that I took my eye off the chat, and only an hour or so later noticed that 1 or 2 messages have arrived that I have never noticed, as there was no sound, and the screen timed out, and I thought that me and the other contact stopped typing only to see that they said something else and I never replied as I didn't notice the messages


Now, I quite agree that a majority of people might not have a need for this, and that is fine. However I am positive that a LOT of other people would find this useful, as nearly all my friends on android tell me the same. Basicaly, the mere fact that both Whatsapp and Viber actually work exactly like this, and furthermore, *that Telegram for iPhones also has this feature already* should pretty much speak for itself.

I believe it should be very easy to implement this, and quiet honestly, I cannot think of a good reason not to. I think Telegram is much better than Whatsapp, but missing this feature is very annoying.

**Comment by Alirezaaa** [comment]
@Darkwood77 Thanks for your long explaining. I've changed my opinion and now I think this feature might be good for some people.

**Comment by wp9015362** [comment]
Please add this feature to Telegram-Android (as an option).

Especially when Telegram-iOS already has this feature.

TextSecure on Android also does have this feature by the way. There is an "In-thread notifications" checkbox inside the settings menu and it's description says "Play notification sound when viewing an active conversation.".

Please also add such an option to Telegram-Android. It would be very useful and much appreciated.

**Comment by AUDI-A4** [comment]
This is right.This option should be present. I don't know why developer is not caring about the what user saying.I am also missing this option.Please add.

**Comment by wp9015362** [comment]
PS:

I have started using Telegram on Android not too long ago, so haven't used it for a long time yet.

It's really great so far.

But this issue (no in-thread notifications) can get really annoying.

Could you please give this a little higher priority and implement it?

But please make it optional.

It would be much appreciated.

Regards

**Comment by AUDI-A4** [comment]
please @DrKLO  add in the next version and add it as optional like whatsapp

**Comment by wp9015362** [comment]
Looks like this is a duplicate of:

https://github.com/DrKLO/Telegram/issues/549

**Comment by wp9015362** [comment]
This is now available in the new Telegram for Android version 2.6.1 from Settings -> Notifications and Sounds -> In-app notifications -> In-Chat Sounds.

But it seems to be buggy as mentioned in:

https://github.com/DrKLO/Telegram/issues/953
